code/Chapter5/test_dependency_injection.cpp

```diff
@@ -0,0 +1,84 @@
+#include <iostream>
+#include <cmath>
+#include "gtest/gtest.h"
+
+/// @brief A "complex" calculation of the norm.
+/// @param array The vector to calculate the norm for.
+/// @return The actual norm of the vector.
+double calculate_norm(std::vector<int> array)
+{
+  double output{0};
+  for (int i{0}; i < array.size(); ++i)
+  {
+    output = array[i] * array[i];
```

Comment:
I think you forgot to add the result to output. In this case, output will always store the square of last element of the array.
```suggestion
    output += array[i] * array[i];
```

Reply:
You're correct.

Reply:
Fixed in 1781f7d